### PR TITLE
[MOS-411] Wired headset audio fix

### PR DESCRIPTION
**Pull request by Lefucjusz**
**Description**

Fixes non-working wired headset audio
during call.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

**Review by SP2FET — APPROVED** [review]
Verified on 4-pole jack headset, audio works in 2-way call

File: products/PurePhone/services/evtmgr/WorkerEvent.cpp [hunk 1]
```diff
@@ -51,16 +51,21 @@ bool WorkerEvent::handleMessage(std::uint32_t queueID)
...
             bsp::headset::HeadsetState::Changed) {
 
-            auto message = std::make_shared<AudioEventRequest>(audio::EventType::JackState,
-                                                               headsetState ? audio::Event::DeviceState::Connected
-                                                                            : audio::Event::DeviceState::Disconnected);
-            service->bus.sendUnicast(message, service::name::evt_manager);
+            auto jackStateMessage = std::make_shared<AudioEventRequest>(
+                audio::EventType::JackState,
+                headsetState ? audio::Event::DeviceState::Connected : audio::Event::DeviceState::Disconnected);
+            service->bus.sendUnicast(jackStateMessage, service::name::evt_manager);
```
Comment (SP2FET) [inline products/PurePhone/services/evtmgr/WorkerEvent.cpp:63]:
```suggestion
            service->bus.sendUnicast(std::move(jackStateMessage), service::name::evt_manager);
```

File: products/PurePhone/services/evtmgr/WorkerEvent.cpp [hunk 2]
```diff
@@ -51,16 +51,21 @@ bool WorkerEvent::handleMessage(std::uint32_t queueID)
...
+
+            auto micStateMessage = std::make_shared<AudioEventRequest>(
+                audio::EventType::MicrophoneState,
+                microphoneState ? audio::Event::DeviceState::Connected : audio::Event::DeviceState::Disconnected);
+            service->bus.sendUnicast(micStateMessage, service::name::evt_manager);
```
Comment (SP2FET) [inline products/PurePhone/services/evtmgr/WorkerEvent.cpp:68]:
```suggestion
            service->bus.sendUnicast(std::move(micStateMessage), service::name::evt_manager);
```

File: module-audio/Audio/AudioCommon.hpp [hunk 3]
```diff
@@ -95,7 +95,8 @@ namespace audio
     enum class EventType
     {
         // HW state change notifications
-        JackState,               //!< jack input plugged / unplugged event
+        JackState, //!< jack input plugged / unplugged event
+        MicrophoneState,
```
Comment (SP2FET) [inline module-audio/Audio/AudioCommon.hpp:99]: please add a comment as in rest of the events :)

File: products/PurePhone/services/evtmgr/WorkerEvent.cpp (hunk 1 above)
Comment (Lefucjusz) [reply products/PurePhone/services/evtmgr/WorkerEvent.cpp]: Changed!

File: products/PurePhone/services/evtmgr/WorkerEvent.cpp (hunk 2 above)
Comment (Lefucjusz) [reply products/PurePhone/services/evtmgr/WorkerEvent.cpp]: Also changed! :P

File: module-audio/Audio/AudioCommon.hpp (hunk 3 above)
Comment (Lefucjusz) [reply module-audio/Audio/AudioCommon.hpp]: Typical "I'm gonna do it later" mistake - added ;)

**Review by mjan-dev — APPROVED** [review]
LGTM

**Pull request merged**